nano/node/node.cpp

```diff
@@ -2906,6 +2907,28 @@ void nano::active_transactions::request_confirm (std::unique_lock<std::mutex> &
 					}
 					election_l->update_dependent ();
 				}
+				// Recalculate work for local blocks if difficulty is below active_difficulty
+				if (election_l->local_block && !election_l->ongoing_work_update)
+				{
+					auto block_l (election_l->status.winner);
+					uint64_t difficulty (0);
+					nano::work_validate (*block_l, &difficulty);
+					if (difficulty < active_difficulty)
+					{
+						election_l->ongoing_work_update = true;
+						auto node_l (node.shared ());
+						auto callback = [node_l, block_l, election_l](boost::optional<uint64_t> const & work_a) {
+							if (work_a && !nano::work_validate (block_l->root (), work_a.get ()))
+							{
+								block_l->block_work_set (work_a.get ());
```

Comment:
This updates a block's work field without a lock / it's not atomic.
Because of the block uniquer, I believe the same block can be accessed or have its work validated by other threads (such as the same block arriving again over bootstrap or realtime traffic)